Buttons interaction

opjs105rogue opened the pull request:
Buttons interaction FIRST STAP (not necessarily for viewing)

1. Created a button to save the file. Linked the renderer.js file to html

2. Created renderer.js because, as i understand it, in Electron applications consist of a main process that controls the application and one or more renderer processes that handle the user interface. Therefore, in order to reduce the likelihood of errors, I separated the main process from the rendering process of processing the button in the html file.
In this file i made an event handler by clicking on the Save button in the application (receives the element specified in the textarea in the application and sends it to the function that we created in preload.js. After that, this function goes to main.js via IPC)

3. In the main.js added an ipcMain moudle that reacts to the "save-text" event that is sent from the renderer. When this event occurs, the code writes the text to a file

opjs105rogue committed:
Buttons interaction: step one

opjs105rogue committed:
add message if file has been saved

opjs105rogue committed:
install sjcl library for crypt/encrypt and add import in js file

opjs105rogue committed:
Now whe you open app, saved text will be show

opjs105rogue committed:
Add interaction with ectript-button

opjs105rogue commented:
@agoalofalife Hi!
In the files I changed, I added the following:

The information recorded in textarea can be saved via the "save" button.

Now, when you open the application, the text that you saved in the application is visible in textarea.

The sjcl library has been added and initialized into project files + imported into main.js .

Added interaction with the "Encrypt" button. Now, when you click this button, an encrypted file is created. (our unencrypted file is not being deleted yet).

Сourse, this is not the final version, so if there are bugs and suggestions, please let me know

opjs105rogue committed:
Edit logic with password and text for SJCL library

opjs105rogue commented:
@agoalofalife , hi!
Added a variable to store our created password.

Because of this, I made corrections and removed unnecessary changes from previous ones. Now, for the SJCL library, the password is transmitted separately from the first window, and the encryption text from the second window

opjs105rogue committed:
Fixed conflicts with main branch

opjs105rogue committed:
delete encrypt function in app for start to do it with new main file

opjs105rogue committed:
delete variable for save pass for sjcl library

opjs105rogue committed:
add button for encrypt and function for encrypt

opjs105rogue committed:
assigned handlers for save and encrypt one one button

opjs105rogue commented:
@agoalofalife , hi!

Fixed conflicts with the main branch.

Added a button for encrypting and saving text in the app. Now, when you click on the button, the information is saved to a file and an encrypted file of our saved txt file is created. (so far, the main txt file is not deleted, I think you need to set a separate task for this, since the implementation of this feature is quite voluminous)

When the application is opened, the information written in the txt file is read and transmitted to the application

opjs105rogue committed:
delete my notes

opjs105rogue commented:
@agoalofalife forgot delete my notes, fixed

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+  
+  //read file and show it in app
+  ipcMain.handle('request-load-text', () => {
+    return fs.readFileSync(filePath, "utf8");
```
I think, it's better to change this string to const and add const to up of this file.
```
const ENCODING = 'utf8'
```

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+      const encryptedText = sjcl.encrypt(hashedPassword, text);
+  
+      fs.writeFileSync(encryptedFilePath, encryptedText, 'utf8');
+      console.info(`Encrypted file saved at: ${encryptedFilePath}`);
```
it's more grammar right way `Encrypted file has been saved to ${encryptedFilePath}`
@RGamz am i right?
and I think, we have to add time log to this line for example
`Encrypted file has been saved to ${encryptedFilePath} at 2024-11-01 12:12:34` <- datetime string format
This log will be more useful 👆

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+  
+      const encryptedText = sjcl.encrypt(hashedPassword, text);
+  
+      fs.writeFileSync(encryptedFilePath, encryptedText, 'utf8');
```
is it possible to use hash instead real password?

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+      const passwordData = getMasterPassword();
+      const { hashedPassword } = passwordData;
+  
+      const encryptedText = sjcl.encrypt(hashedPassword, text);
```
is it possible to use hash instead real password?

opjs105rogue replied:
Yes, the password hash is passed to the library instead of the password itself

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+    const encryptedFilePath = path.join(userFilesDir, 'encrypted.txt');
+  
+    try {
+      fs.writeFileSync(filePath, text, 'utf8');
```
it doesn't make sense, save not encrypted plain text in file, always save only encrypted

opjs105rogue replied:
Based on this, I think that the
//read file and show it in app part
 ipcMain.handle('request-load-text', () => {
 return fs.readFileSync(filePath, ENCODING);
 });
In the main.js also don't need, but request-load-text from preload.js and renderer.js  should not delete, it will be useful when we read information from the decrypted file

agoalofalife commented on main.js:
```diff
@@ -90,6 +89,48 @@ const renderMainWindow = () => {
+        message: 'The text has been saved and encrypted successfully!'
+      });
+    } catch (error) {
+      console.error('Error during save and encryption:', error);
```
it's better to add date time which i mentioned before, and of course it has to be special function to generate date

opjs105rogue committed:
cosmetic changes

opjs105rogue commented:
Hi, @agoalofalife 

Changed the format of the folder name to camel notation.

Removed unnecessary functions when reading and saving an unencrypted file.

Created a function that returns the date in the specified format.

Added date functions to logs and notifications.

Const ENCODING was created for code flexibility.

RGamz commented on main.js:
```diff
@@ -90,6 +90,54 @@ const renderMainWindow = () => {
     }
   })();
+  
+  function getDatаNow() {
```
@opjs105rogue  rename this function to "getCurrentDatetime()", because data refers to information and not to date.

opjs105rogue committed:
Fix function name

opjs105rogue commented:
renamed name function for get Data

opjs105rogue committed:
deleted an unused variable

opjs105rogue committed:
add moment.js and fixed time function

opjs105rogue commented:
Add moment.js and edit data function

agoalofalife committed:
small fixes

opjs105rogue committed:
small fix logic with create/delete unencrypt txt file

opjs105rogue committed:
Fix #2 and add function for encrypt\decrypt for code's flexibility

opjs105rogue committed:
small fixes

opjs105rogue committed:
SMALL FIXES TRY 2

opjs105rogue commented:
@agoalofalife and @RGamz hi!
Added a separate function for encryption and decryption for code flexibility.

Used a handler to read the "request-load-text" files, and now there is no error in the application console anymore.

Added a variable to .env for code flexibility.

Slightly changed the logic of creating a file and reading it. Now, when you first log into the application, a txt file is created, when you click on the "Save&Encrypt" button, it is saved and encrypted into a separate file (which we will work with further), and deleted.
On subsequent logins to the application, we enter the password and receive decrypted information that we can change

opjs105rogue committed:
fix log info for user frendly

opjs105rogue committed:
Redesign of the file creation and encryption

opjs105rogue commented:
Hi, @agoalofalife @RGamz !

Deleted unnecessary variables.

Logs have been edited in the absence of a variable in .env.

The logic of creating and encrypting information in a file has been changed. Now the text is encrypted first and a file with this encrypted text is created when you click on the button in the main menu.
